pkg/controller/controller.go

```diff
@@ -739,6 +780,65 @@ func findServicePlan(name string, plans []v1alpha1.ServicePlan) *v1alpha1.Servic
 	return nil
 }
 
+// handleLastOperationResponse gets called when a last operation poll has happened.
+func (c *controller) handleLastOperationResponse(instance *v1alpha1.Instance, resp *brokerapi.LastOperationResponse) bool {
+	switch resp.State {
+	case "in progress":
+		c.setAsyncOperationOngoing(instance, true)
+		return true
+	case "succeeded":
+		c.setAsyncOperationOngoing(instance, false)
+		// TODO(vaikas): Update the instance condition ready to true and set to succeed
+	case "failed":
+		c.setAsyncOperationOngoing(instance, false)
+		// TODO(vaikas): Update the instance condition ready to true and set to fail
+	default:
+		glog.Warningf("Got invalid state in LastOperationResponse: %q", resp.State)
+	}
+	return false
+}
+
+// setAsyncOperationOngoing updates the
+// InstanceConditionAsyncOperationInProgress to the specified status.
+func (c *controller) setAsyncOperationOngoing(instance *v1alpha1.Instance, status bool) {
+	if status {
+		c.updateInstanceCondition(
```

Comment:
My opinion is in-line with @vaikas-google's - having another condition adds more programmatically accessible information on the state of the system.

Regardless, I would like to get beyond this so that @vaikas-google can move forward. Please comment if you are significantly opposed to the following:

- Adding a `AsyncOperationInProgress` condition
- Setting that condition on the instance after the broker returns an async response
- Setting `InstanceConditionReady` after the async operation completes